src/mechanics/constraintsPde/ConstraintCompanion.cpp

```diff
@@ -0,0 +1,125 @@
+#include "mechanics/constraintsPde/ConstraintCompanion.h"
+#include "mechanics/nodes/NodeSimple.h"
+#include "mechanics/constraintsPde/Equation.h"
+
+namespace NuTo
+{
+namespace ConstraintPde
+{
+
+std::vector<Equation> Component(const NodeSimple& node, std::vector<eDirection> directions, double value)
+{
+    return Component(node, directions, RhsConstant(value));
+}
+
+
+std::vector<Equation> Component(const NodeSimple& node, std::vector<eDirection> directions, RhsFunction rhs)
+{
+    std::vector<Equation> eqs;
+    for (auto direction : directions)
+    {
+        int component = ToComponentIndex(direction);
+        eqs.emplace_back(node, component, rhs);
+    }
+    return eqs;
+}
+
+
+std::vector<Equation> Component(const Groups::Group<NodeSimple>& nodes, std::vector<eDirection> directions,
+                                double value)
+{
+    return Component(nodes, directions, RhsConstant(value));
+}
+
+
+std::vector<Equation> Component(const Groups::Group<NodeSimple>& nodes, std::vector<eDirection> directions,
+                                RhsFunction rhs)
+{
+    std::vector<Equation> eqs;
+    for (const auto& node : nodes)
+    {
+        auto tmpEqs = Component(node, directions, rhs);
+        eqs.insert(eqs.end(), tmpEqs.begin(), tmpEqs.end());
+    }
+    return eqs;
+}
+
+
+Equation Direction(const NodeSimple& node, Eigen::VectorXd direction, RhsFunction rhs)
+{
+
+    assert(direction != Eigen::VectorXd::Zero(direction.rows()) && "the direction vector contains only zeros");
```

Comment:
It is always a bit risky to compare floating point numbers without a tolerance. If you change that assert into an actual exception, you could properly test it.

Reply:
Yep I know. However, since the vector is normalized even the tiniest number should be enough to get everything back on track. But I ll replace that with a tolerance. 
Something like: max(abs(v)) > 1e-6;